lockfile/src/lib.rs

```diff
@@ -140,7 +143,7 @@ impl LockfileFormat {
 }
 
 /// An iterator of all supported lockfile formats.
-pub struct LockfileFormatIter(u8);
+pub struct LockfileFormatIter(usize);
```

Comment:
Instead of `usize` here, it could be `std::slice::Iter<'static, LockfileFormat>`. The size of the struct increases by 16x instead of 8x like this code, but `next` would become `Some(*self.0.next()?)`.